cpg_workflows/jobs/simple_vep_annotation.py

```diff
@@ -0,0 +1,206 @@
+    bcftools_job = get_batch().new_job('Subset VCF with bcftools')
+
+    # set some resources
+    bcftools_job.image(image_path('bcftools')).cpu(1).memory('8G')
+
+    # todo storage? VCF * 2.5
+
+    # existing fragments, instead of checking for each separately. This is an empty list if the 'dir' doesn't exist yet
+    existing_fragments = [str(each_path) for each_path in to_path(output_dir).glob('*')] if output_dir else []
+    logging.info(f'Existing VCFs: {existing_fragments}')
+
+    for chrom in CHROM_LIST:
+
+        # if a location to write to was used, check if we already made this (workflow resumption)
+        if isinstance(output_dir, str):
+            # the name for this chunk of annotation
+            result_path = os.path.join(output_dir, f'{chrom}.vcf.bgz')
+
+            # check if it exists already, if so read it in
+            if result_path in existing_fragments:
+                logging.info(f'{result_path} already exists')
+                vcf_fragment = get_batch().read_input_group(
+                    **{
+                        'vcf.bgz': result_path,
+                        'vcf.bgz.tbi': f'{result_path}.tbi',
+                    },
+                )['vcf.bgz']
+                ordered_output_vcfs.append(vcf_fragment)
+                continue
+
+        # otherwise declare a new resource group, with a name exclusive to this chromosome
+        bcftools_job.declare_resource_group(
+            **{chrom: {'vcf.bgz': '{root}.vcf.bgz', 'vcf.bgz.tbi': '{root}.vcf.bgz.tbi'}},
+        )
+
+        # create a VCF fragment for this chromosome
+        bcftools_job.command(
+            f'bcftools view -Oz -o {bcftools_job[chrom]["vcf.bgz"]} -r {chrom} {localised_vcf}',  # type: ignore
+        )
+
+        # index it
+        bcftools_job.command(f'tabix {bcftools_job[chrom]["vcf.bgz"]} ')  # type: ignore
+
+        if isinstance(output_dir, str):
+            # write the fragment & index to GCP
+            get_batch().write_output(bcftools_job[chrom], os.path.join(output_dir, chrom))
+
+        # and add to the sorted list for this batch
+        ordered_output_vcfs.append(bcftools_job[chrom]['vcf.bgz'])  # type: ignore
+
+    return ordered_output_vcfs
+
+
+def annotate_localised_vcfs(vcf_list: list[ResourceFile], output_dir: str | None = None) -> list[ResourceFile]:
+    """
+    annotate each VCF fragment using VEP. Optionally attempt to resume from a folder, and write results to same
+
+    Args:
+        vcf_list ():
+        output_dir ():
+
+    Returns:
+        a list of localised-to-batch annotated VCFs
+    """
+
+    # existing outputs
+    existing_outputs = [str(each_path) for each_path in to_path(output_dir).glob('*')] if output_dir else []
+
+    logging.info(f'Existing annotated VCFs: {existing_outputs}')
+
+    ordered_annotated: list = []
+
+    # next, annotate!
+    for chrom, vcf in zip(CHROM_LIST, vcf_list, strict=True):
```

Comment:
I've gotten rid of this, so that this method can take any number of input VCFs. It'll just annotate whatever it's given and write out as `1.vcf`, `2.vcf`